zoo/Cage.java

```diff
@@ -0,0 +1,169 @@
+package zoo;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Cage {
+	
+	/***********Instance Variables********************************/
+	
+	int capacity;//number of animal capacity in a particular cage
+        int numberOfAnimalInCage;//number of animal present in cage
+
+	Animal animalType;//specific type of animal like Lion,Tiger,Dog,etc.
+	
+	List<Animal> animal;//list of animal in this cage
+	
+	
+	
+	
+	
+	
+	/***********************Constructor***************************/
+	/*
+	 * @param capacity ,maximum number of animal who can reside in a cage
+	 * @param animalType, specific animal type which will be in cage like Lion,Peacock etc
+	 * */
+	public Cage(int capacity, Animal animalType){
+		this.capacity = capacity;
+		this.animalType = animalType;
+		this.animal = new ArrayList<Animal>();
+		
+	}
+	
+	
+	
+	/************************public Methods*************************/
+	
+	/*
+	 * @param animal , animal which is going to be add in this cage
+	 * @return "true" if animal successfully added else "false"
+	 * */
+	public boolean addAnimal(Animal addAnimal){
+		boolean isAdded=false;
+		
+		//checking all basic condition before addition
+		if(isAllConditionIsTrue(addAnimal)){
+			
+				//if number of element in cage is under capacity
+				if(this.animal.size()<this.capacity){
```

Comment:
We shall use Array instead of ArrayList.
After using Array, this capacity check shall be removed.